src/Components/DevPosition.jsx

```diff
@@ -3,54 +3,28 @@ import "../App.css";
 
 export default function DevPosition({ devString, updatePosition }) {
   const [currentString, setCurrentString] = useState("");
-  //   const [ind, setInd] = useState(null);
-  const [didRun, setDidRun] = useState(false);
-  const [reachEnd, setReachEnd] = useState(false);
+  const [ind, setInd] = useState(null);
 
-  const updateDevString = () => {
-    if (!didRun) {
-      setDidRun(true);
-    } else {
-      setTimeout(() => {
-        setCurrentString((prevState) => {
-          if (reachEnd && prevState.length === 1) {
-            console.log("updating");
-            updatePosition(curr => (curr + 1) % 3);
-            setDidRun(false);
-            setReachEnd(false);
-            return "";
-          } else if (prevState.length >= devString.length || reachEnd) {
-            setReachEnd(true);
-            return devString.slice(0, prevState.length - 1);
-          } else { 
-            return devString.slice(0, prevState.length + 1);
-          }
-        });
-      }, 500);
-    }
-    // setTimeout(() => {
-    //   setCurrentString(() =>
-    //     devString.slice(0, devString.length - Math.abs(ind))
-    //   );
-    //   setInd((prevInd) => prevInd - 1);
-    // }, 200);
-  };
+  useEffect(() => {
+    setInd(devString.length);
+  }, [devString]);
 
   useEffect(() => {
-    updateDevString();
-  }, [currentString, didRun]);
+    if (ind === null) return;
 
-  //   useEffect(() => {
-  //     setInd(devString.length);
-  //   }, [devString]);
+    setCurrentString(() =>
+      devString.slice(0, devString.length - Math.abs(ind))
+    );
 
-  //   useEffect(() => {
-  //     if (ind === -devString.length) {
-  //       setInd(null);
-  //       updatePosition((prevPos) => (prevPos >= 2 ? 0 : prevPos + 1));
-  //     }
-  //     updateDevString();
-  //   }, [currentString]);
+    if (ind === -devString.length) {
+      setInd(null);
+      updatePosition((prevPos) => (prevPos >= 2 ? 0 : prevPos + 1));
+    } else {
+      setTimeout(() => {
+        setInd((prevInd) => prevInd - 1);
+      }, 200);
```

Comment:
I have combined the separate function that we had for timeout in the `useEffect` itself. Let me know if this should stay separate. I'll update if you want it how it was before.